src/apps/btc/btc_common.c

```diff
@@ -359,3 +359,33 @@ bool btc_common_pkscript_from_outputhash(
     }
     return true;
 }
+
+bool btc_common_xpub_from_protobuf(const XPub* xpub_in, struct ext_key* xpub_out)
+{
+    struct ext_key* tmp = NULL;
+    // libwally quirk: bip32_key_init_alloc takes 20 bytes for the parent fingerprint but only uses
+    // the first 4 bytes (the real size of the fingerprint).
+    uint8_t parent_fingerprint[20] = {0};
+    memcpy(parent_fingerprint, xpub_in->parent_fingerprint, sizeof(xpub_in->parent_fingerprint));
+    // Another libwally quirk: no function to create an ext_key from params without malloc.
```

Comment:
did you report these issues to libwally repo?

Reply:
Just did. I have a whole bunch of issues, I think we need to provide the PRs ourselves though.